### PR TITLE
mod flow implementation with bug fixes

### PR DESCRIPTION
**Pull request by faebluna**

### CODE REVIEW

File: DiscordBot/moderator.py [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+            
+            if i == 0:
+                await self.handle_escalate(report)
+                await self.handle_ban(report)
+                self.state = ModeratorState.ACTION_COMPLETE
+
+            if i == 1:
+                self.state = ModeratorState.AWAITING_HIDE_CHILDREN_DECISION
+                # TODO increase number of times user has been reported
+                reply = self.create_options_list("We increased the number of false reports associated to this user. Would you like to hide child profiles from this user?",
```
Comment (larafranciulli) [inline DiscordBot/moderator.py]: We need to change the text to be: 

`We increased the number of times this user has been reported for CSAM. Would you like to hide child profiles from this user?`

File: DiscordBot/moderator.py [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+    LEGIT_REPORT_DECISION_OPTIONS = [
+        "Escalate to law enforcement",
+        "It does not violate policy/the law, but not appropriate behavior (e.g., text a minor saying they are cute)"
+    ]
+    FALSE_REPORT_DECISION_OPTIONS = [
+        "Warn user about not submitting false reports",
+        "Increase the number of times user has submitted false reports"
+    ]
+    REPORT_FREQUENCY_DECISION_OPTIONS = [
+        "Report to NCMEC",
```
Comment (larafranciulli) [inline DiscordBot/moderator.py]: Should we just change this to law enforcement and prompt the user to report to both NCMEC and Lantern?

Something like:

"Please report to NCMEC here and Lantern here. When you are done with the reporting, type Yes"

File: DiscordBot/bot.py [hunk 3]
```diff
@@ -108,20 +113,41 @@ async def handle_channel_message(self, message):
...
+            #     # Call the handle_ban function from the ModeratorReport instance
+            #     await moderator_report.handle_ban(message)
+            # elif message.content.startswith('!hide'):
+            #     # Call the handle_hide_profile function from the ModeratorReport instance
+            #     await moderator_report.handle_hide_profile(message)
+            # elif message.content.startswith('!escalate'):
+            #     # Call the handle_escalate function from the ModeratorReport instance
+            #     await moderator_report.handle_escalate(message)
+            # else:
+            #     await message.channel.send("Please enter a valid command: `!ban`, `!hide`, or `!escalate`.")
```
Comment (larafranciulli) [inline DiscordBot/bot.py]: Should we remove the comments and prints before we merge?

**Review by larafranciulli — CHANGES_REQUESTED** [review]
I left some comments that we might want to change before merging it!

File: DiscordBot/moderator.py (hunk 2 above)
Comment (faebluna) [reply DiscordBot/moderator.py]: By user do you mean the mod going through the mod flow or the user who made the report? I think if its still going to be up to the mod to report to NCMEC and Lantern we should just leave it as an option and assume that by picking the option the reports are made by the system

File: DiscordBot/bot.py (hunk 3 above)
Comment (faebluna) [reply DiscordBot/bot.py]: removed this and other comments and print statements I found

File: DiscordBot/moderator.py (hunk 1 above)
Comment (faebluna) [reply DiscordBot/moderator.py]: Changed.

**Review by larafranciulli — APPROVED** [review]

File: DiscordBot/moderator.py (hunk 2 above)
Comment (larafranciulli) [reply DiscordBot/moderator.py]: Makes sense!

**Pull request merged**